core/DataAccess/TableMetadata.php

```diff
@@ -0,0 +1,56 @@
+<?php
+/**
+ * Piwik - free/libre analytics platform
+ *
+ * @link http://piwik.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ */
+namespace Piwik\DataAccess;
+
+use Piwik\Db;
+
+/**
+ * Data Access Object that can be used to get metadata information about
+ * the MySQL tables Piwik uses.
+ */
+class TableMetadata
+{
+    /**
+     * Returns the list of column names for a table.
+     *
+     * @param string $table Prefixed table name.
+     * @return string[] List of column names..
+     */
+    public function getColumns($table)
+    {
+        $table = str_replace("`", "", $table);
+
+        $columns = Db::fetchAll("SHOW COLUMNS FROM `" . $table . "`");
```

Comment:
I'm attempting to avoid SQL injections here (since getColumns is a public method, I am assuming its possible for plugin developers to do stupid things w/ it).